lib/oban/postgres_notifier.ex

```diff
@@ -0,0 +1,201 @@
+defmodule Oban.PostgresNotifier do
+  @doc """
+  Postgres Listen/Notify based Notifier
+  """
+
+  @behaviour Oban.Notifier
+
+  use GenServer
+
+  import Oban.Breaker, only: [open_circuit: 1, trip_circuit: 3]
+
+  alias Oban.{Config, Registry, Repo}
+  alias Postgrex.Notifications
+
+  @mappings %{
+    gossip: "oban_gossip",
+    insert: "oban_insert",
+    signal: "oban_signal"
+  }
+
+  @channels Map.keys(@mappings)
+
+  defmodule State do
+    @moduledoc false
+
+    @enforce_keys [:conf]
+    defstruct [
+      :conf,
+      :conn,
+      :name,
+      circuit: :enabled,
+      reset_timer: nil,
+      listeners: %{}
+    ]
+  end
+
+  defguardp is_channel(channel) when channel in @channels
+
+  @impl Oban.Notifier
+  def start_link(opts) do
+    name = Keyword.get(opts, :name, __MODULE__)
+
+    GenServer.start_link(__MODULE__, Map.new(opts), name: name)
+  end
+
+  @impl Oban.Notifier
+  def listen(server \\ Oban, channels)
+
+  def listen(pid, [_ | _] = channels) when is_pid(pid) do
+    :ok = validate_channels!(channels)
+
+    GenServer.call(pid, {:listen, channels})
+  end
+
+  def listen(oban_name, [_ | _] = channels) do
+    oban_name
+    |> Registry.whereis(Oban.Notifier)
+    |> listen(channels)
+  end
+
+  @impl Oban.Notifier
+  def unlisten(oban_name \\ Oban, [_ | _] = channels) do
```

Comment:
Similarly to `listen/2`, we can simplify this to be `unlisten(server, channels)`.